VineTime - Gruppe A

Tessellation93 opened the pull request:
VineTime Pomodoro er en produktivitetstimer som følger Pomodoro teknikken, hvor man arbejder i intervaller af 25 minutter og holder 5 eller 10 min pause.

Commit:
idc

Commit:
added vineTimeController routes.

Commit:
Vinetime - controller route added with demo login

Commit:
Vinetime - updated index.html with login

Commit:
Vinetime - added sql file

Commit:
Vinetime - index.html corrected and working

Commit:
Vinetime - DAO updated to save sessions with duration and exceptions

Commit:
Vinetime - corrected table and column names deleted accidental ´s´

Commit:
Vinetime - frontend rebuild after Figma issue

I built the VineTime design in Figma myself and tried to connect it with React at first. Because the project setup was with Thymeleaf, I used Claude to help me convert and restyle everything in plain HTML, CSS, and JS. Learnt a lot about frameworkds and how easily things can clash in shared projects.

Commit:
Vinetime - Cleaned up frontend files after the Figma export f-up

Commit:
Vinetime - Images for frontend replaced

Commit:
Vinetime - rebuilt CSS from Figma

See previous notes on js and css file for desciption

Commit:
Vinetime - entity refactor of duration and completedAt

Updated conctructors to match the new parameters. Only naming and structural changes

Commit:
Vinetime - Updated  SessionMapper field names

Matching new field names for concistency

Commit:
Made pgn files smaller så i could create a PR

tjorne commented on src/main/java/app/controllers/VineTimeController.java:
```diff
@@ -0,0 +1,114 @@
+        String username = ctx.formParam("username") != null ? ctx.formParam("username") : "";
+        String password = ctx.formParam("password") != null ? ctx.formParam("password") : "";
+
+        if ("test".equals(username) && "1234".equals(password)) {
```
test logins bør ikke pushes op på den fælles kode base.

tjorne commented on src/main/java/app/controllers/VineTimeController.java:
```diff
@@ -0,0 +1,114 @@
+        ctx.render("/vinetime/index.html");
+    }
+
+    private static void login(Context ctx) {
```
hvorfor ikke benytte den eksisterende login funktionalitet i stedet?

tjorne commented on src/main/java/app/persistence/ConnectionPool.java:
disse ændringer bør ikke være med. de er specifikke for dig/jer og dermed bør de ikke deles med alle andre. 
Vær venlig at revert'e ændringer til denne fil.

tjorne commented on src/main/java/app/persistence/VineTimeSessionDAO.java:
nice

tjorne commented on src/main/resources/config_vinetime.properties:
jeg tænker heller ikke den her fil er relevant at have i den fælles løsning? Vær venlig at fjerne den igen.

tjorne commented on src/main/resources/public/js/vinetime.js:
```diff
@@ -0,0 +1,55 @@
+            e.preventDefault();
+            const u = document.getElementById('email');
+            const p = document.getElementById('password');
+            if (u) u.value = 'test';
```
hmm, virker igen som en overflødig test tilføjelse - i hvert fald set i perspektivet af den store master branch der skal holdes ren.

tjorne commented on src/main/resources/sql/vinetime.sql:
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE users (
```
konflikterende tabel navn - vi har allerede en users tabel i projektet.

tjorne reviewed (changes requested):
Det ser rigtig rigtig fint ud det her! 

Jeg har tilføjet en håndfuld kommentarer til dette PR, som du/I meget gerne må udbedre før ændringerne / tilføjelserne kan komme med ind i master branch'en.

tjorne commented:
Oh, og så må du gerne merge master ind i din branch igen! Der er et par enkelte conflicts som bør tages på jeres egen branch.

tjorne commented:
> Complete VineTime Pomodoro timer.

Normalt ser vi også gerne en fin beskrivelse til et PR, med hvad det indeholder, hvad man kan forvente at finde, eventuelt instrukser til hvordan man kører det, osv..

Commit:
Merge branch 'master' of https://github.com/tjorne/lifehackE25 into gruppeA/dev

Commit:
Rettelser efter review

- Har fjernet det overflødige lokale og mine testfiler, Mapper, SQL og config)
- Opdateret VinetimeController for at passe til den delte session
- Fjernet test login
- Fjernet ændringer i delte kode

Tessellation93 reviewed (commented):
Final review
